sticker-graph/sticker_graph/write_helper.py

```diff
@@ -17,29 +17,33 @@ def create_graph(model, args):
     shapes = read_shapes(args)
     graph_filename = args.output_graph_file
 
-    gpuopts = tf.GPUOptions(per_process_gpu_memory_fraction=0.3)
-    tfconfig = tf.ConfigProto(gpu_options=gpuopts)
+    gpuopts = tf.compat.v1.GPUOptions(per_process_gpu_memory_fraction=0.3)
+    tfconfig = tf.compat.v1.ConfigProto(gpu_options=gpuopts)
 
-    with tf.Graph().as_default(), tf.Session(config=tfconfig) as session:
+    with tf.Graph().as_default(), tf.compat.v1.Session(config=tfconfig) as session:
         write_args(model, args)
 
-        logdir = tf.placeholder(shape=[], name="logdir", dtype=tf.string)
-        summary_writer = sticker_graph.vendored._create_file_writer_generic_type(logdir)
+        logdir = tf.compat.v1.placeholder(
+            shape=[], name="logdir", dtype=tf.string)
+        summary_writer = sticker_graph.vendored._create_file_writer_generic_type(
+            logdir)
 
         with summary_writer.as_default(), tf.contrib.summary.always_record_summaries():
-            with tf.variable_scope("model", reuse=None):
+            with tf.compat.v1.variable_scope("model", reuse=None):
                 model(args=args, shapes=shapes)
 
             tf.group(tf.contrib.summary.summary_writer_initializer_op(),
                      name="summary_init")
-            tf.variables_initializer(tf.global_variables(), name='init')
+            tf.compat.v1.variables_initializer(
+                tf.compat.v1.global_variables(), name='init')
 
-            tf.train.Saver(tf.global_variables())
+            tf.compat.v1.train.Saver(tf.compat.v1.global_variables())
 
             serialized_graph = session.graph_def.SerializeToString()
             serialized_graph_tensor = tf.convert_to_tensor(serialized_graph)
-            tf.contrib.summary.graph(serialized_graph_tensor, 0, name='graph_write')
-            tf.train.write_graph(
+            tf.contrib.summary.graph(
```

Comment:
`contrib` will be gone in 2.0, need to check if the method has moved elsewhere.

Reply:
`tf_upgrade_v2` gives suggestions for replacing `contrib.summary.graph`, but also mentions that it is not a 1:1 conversion. Maybe you could have a look at this sometime after this PR is merged? (I didn't want to touch the summary-writing parts of the graph.)

Reply:
Should this PR be compatible with 2.0 already?

Reply:
No, it's just one step towards compatibility. E.g. for CRFs we need to introduce tensorflow-addons as a dependency.